python/lsst/ts/mtdome/mtdome_csc.py

```diff
@@ -511,48 +555,64 @@ async def write_then_read_reply(
             TimeoutValue} where "response" can be zero for "OK" or non-zero
             for "ERROR".
         """
-        command_dict = dict(command=command, parameters=params)
-        st = encoding_tools.encode(**command_dict)
+        command_id = next(self._index_iter)
+        self.commands_without_reply[command_id] = CommandTime(
+            command=command, tai=utils.current_tai()
+        )
+        command_dict = dict(commandId=command_id, command=command, parameters=params)
         async with self.communication_lock:
             try:
-                assert self.writer is not None
+                assert self.client is not None
             except AssertionError as e:
-                await self.fault(code=3, report=f"Error writing command {st}: {e}")
+                await self.fault(
+                    code=3, report=f"Error writing command {command_dict}: {e}."
+                )
                 raise
 
             disabled_commands: set[str] = set()
             if self.simulation_mode == ValidSimulationMode.NORMAL_OPERATIONS:
                 disabled_commands = COMMANDS_DISABLED_FOR_COMMISSIONING
 
             if command not in disabled_commands:
-                self.writer.write(st.encode() + b"\r\n")
-                await self.writer.drain()
                 try:
-                    assert self.reader is not None
-                    read_bytes = await asyncio.wait_for(
-                        self.reader.readuntil(b"\r\n"), timeout=_TIMEOUT
+                    await self.client.write_json(data=command_dict)
+                    data = await asyncio.wait_for(
+                        self.client.read_json(), timeout=_TIMEOUT
                     )
                 except (
                     asyncio.exceptions.IncompleteReadError,
                     asyncio.exceptions.TimeoutError,
-                    ConnectionResetError,
+                    ConnectionError,
```

Comment:
Do you need to be so specific, or can you just catch Exception? What exceptions are you trying allow to propagate up without going to fault?

Reply:
No need at all. I reduced it to `Exception`.